infra/IExtendedAgent.go

```diff
@@ -23,6 +23,9 @@ type IExtendedAgent interface {
 	IsAlive() bool
 
 	GetTargetPosition() (PositionVector, bool)
+	GetClusterWeightedMeanPosition() PositionVector
+	GetNetworkWeightedMeanPosition() PositionVector
+	NormalizeToUnit(PositionVector) PositionVector
```

Comment:
This is the perfect example of where you should be precise with your OOP. A method that operates on, and returns, a `PositionVector` should exist within the `PositionVector` class. 

At the same time, as mentioned before, the server should and does handle the conversion to a value in the range [-1, 1]